File: frontend/src/api/skills.ts
/**
 * api/skills.ts — M8
 * All skill verification API calls.
 */
import { api } from './client'

export interface SkillCatalogueItem {
  key:            string
  label:          string
  icon:           string
  description:    string
  question_count: number
}

export interface QuestionOut {
  index:   number
  text:    string
  options: string[]
}

export interface TestStartResponse {
  test_id:          string
  skill_key:        string
  skill_label:      string
  questions:        QuestionOut[]
  duration_seconds: number
  started_at:       number
}

export interface TestSubmitRequest {
  answers: Record<string, number>  // "0" → chosen option index
}

export interface TestSubmitResponse {
  verification_id: string
  skill_key:       string
  skill_label:     string
  score_pct:       number
  correct:         number
  total:           number
  badge:           'bronze' | 'silver' | 'gold' | null
  time_taken_s:    number | null
  completed_at:    string
}

export interface VerifiedSkillOut {
  id:           string
  skill_key:    string
  skill_label:  string
  score_pct:    number
  badge_level:  'bronze' | 'silver' | 'gold' | null
  completed_at: string | null
}

export const skillsApi = {
  listTests: (): Promise<SkillCatalogueItem[]> =>
    api.get('/skills/tests').then(r => r.data),

  startTest: (skillKey: string): Promise<TestStartResponse> =>
    api.post(`/skills/tests/${skillKey}/start`).then(r => r.data),

  submitTest: (testId: string, payload: TestSubmitRequest): Promise<TestSubmitResponse> =>
    api.post(`/skills/tests/${testId}/submit`, payload).then(r => r.data),

  getVerified: (): Promise<VerifiedSkillOut[]> =>
    api.get('/skills/verified').then(r => r.data),
}

File: backend/app/schemas/skills.py
"""
Pydantic schemas for M8 skill verification endpoints.
"""
from __future__ import annotations
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


# ── Catalogue ──────────────────────────────────────────────────────────────────

class SkillCatalogueItem(BaseModel):
    key:            str
    label:          str
    icon:           str
    description:    str
    question_count: int


# ── Test start ─────────────────────────────────────────────────────────────────

class QuestionOut(BaseModel):
    index:   int
    text:    str
    options: list[str]


class TestStartResponse(BaseModel):
    test_id:           str
    skill_key:         str
    skill_label:       str
    questions:         list[QuestionOut]
    duration_seconds:  int
    started_at:        float


# ── Test submit ────────────────────────────────────────────────────────────────

class TestSubmitRequest(BaseModel):
    answers: dict[str, int]    # {"0": 2, "1": 0, …}  question_index → chosen_option_index


class TestSubmitResponse(BaseModel):
    verification_id: str
    skill_key:       str
    skill_label:     str
    score_pct:       float
    correct:         int
    total:           int
    badge:           Optional[str]     # "bronze" | "silver" | "gold" | null
    time_taken_s:    Optional[int]
    completed_at:    datetime


# ── Verified skills list ───────────────────────────────────────────────────────

class VerifiedSkillOut(BaseModel):
    id:           str
    skill_key:    str
    skill_label:  str
    score_pct:    float
    badge_level:  Optional[str]
    completed_at: Optional[datetime]

    class Config:
        from_attributes = True
